README.md

```diff
@@ -19,9 +19,408 @@ BiggerQuery lets you implement complex stuff (the Allegro experimentation platfo
+    external_tables={
+        '311_requests': f'{PROJECT_ID}.external_data.311_requests'
+    },
+    internal_tables=['request_aggregate'])
+
+create_table = dataset.create_table("""
+CREATE TABLE IF NOT EXISTS request_aggregate (
+    batch_date TIMESTAMP,
+    request_count INT64)
+PARTITION BY DATE(batch_date)""")
+
+write_truncate_daily_request_count = dataset.write_truncate('request_aggregate', """
+WITH batched_requests as (
+    SELECT 
+        DATE(TIMESTAMP(created_date)) as batch_date,
+        *
+    FROM `{311_requests}`
+    WHERE DATE(TIMESTAMP(created_date)) = "{dt}"
+)
+
+SELECT 
+    TIMESTAMP(batch_date) as batch_date,
+    count(*) as request_count
+FROM `batched_requests`
+WHERE DATE(TIMESTAMP(created_date)) = "{dt}"
+GROUP BY batch_date
+""")
+
+workflow_v1 = bgq.Workflow(definition=[
+    create_table.to_job(),
+    write_truncate_daily_request_count.to_job()
+])
+```
+
+Next, create another notebook and add the following code that will generate the Airflow DAG:
+
+```python
+import biggerquery as bgq
+
+bgq.build_dag_from_notebook('/path/to/your/notebook.ipynb', 'workflow_v1', start_date='2014-05-21')
+```
+
+After you run the code above, you will get [zipped Airflow DAG](https://airflow.apache.org/concepts.html?highlight=zip#packaged-dags) that you can deploy.
+The easiest way to deploy a DAG is by using the [Cloud Composer](https://cloud.google.com/composer/).
+
+### Wait for tables
+
+If you want to wait for some data to appear before you start a processing, you can use sensor component:
+
+```python
+import biggerquery as bgq
+
+PROJECT_ID = 'put-you-project-id-here'
+
+dataset = bgq.Dataset(
+    project_id=PROJECT_ID,
+    dataset_name='biggerquery_cheatsheet',
+    external_tables={
+        '311_requests': f'{PROJECT_ID}.external_data.311_requests'
+    },
+    internal_tables=['request_aggregate'])
+
+wait_for_requests = bgq.sensor_component(
+    '311_requests', 
+    where_clause="DATE(TIMESTAMP(created_date)) = DATE(TIMESTAMP_ADD(TIMESTAMP('{dt}'), INTERVAL -24 HOUR))",
+    ds=dataset)
+
+workflow_v2 = bgq.Workflow(definition=[wait_for_requests.to_job()])
+
+# Should raise ValueError because there is no data for '2090-01-01'
+workflow_v2.run('2090-01-01')
+```
+
+### Write custom component
+
+If you want to write you own component, you can do it by writing a function:
+
+```python
+import biggerquery as bgq
+
+PROJECT_ID = 'put-you-project-id-here'
+
+dataset = bgq.Dataset(
+    project_id=PROJECT_ID,
+    dataset_name='biggerquery_cheatsheet',
+    external_tables={
+        '311_requests': f'{PROJECT_ID}.external_data.311_requests'
+    },
+    internal_tables=['request_aggregate'])
+
+@bgq.component(ds=dataset)
+def wait_for_requests(ds):
+    result = ds.collect('''
+        SELECT count(*) > 0 as table_ready
+        FROM `{311_requests}`
+        WHERE DATE(TIMESTAMP(created_date)) = DATE(TIMESTAMP_ADD(TIMESTAMP('{dt}'), INTERVAL -24 HOUR))
+        ''')
+
+    if not result.iloc[0]['table_ready']:
```

Comment:
Could we create method for this? `is_table_ready`?

Reply:
done